clients/snowflake/dialect/dialect.go

```diff
@@ -225,20 +225,28 @@ MERGE INTO %s %s USING ( %s ) AS %s ON %s`,
 		tableID.FullyQualifiedName(), constants.TargetAlias, subQuery, constants.StagingAlias, strings.Join(equalitySQLParts, " AND "),
 	)
 
+	cols, removed := columns.RemoveOnlySetDeleteColumnMarker(cols)
+	if !removed {
+		return []string{}, errors.New("artie only_set_delete flag doesn't exist")
+	}
+
 	if softDelete {
 		return []string{baseQuery + fmt.Sprintf(`
-WHEN MATCHED %sTHEN UPDATE SET %s
+WHEN MATCHED %sAND IFNULL(%s, false) = false THEN UPDATE SET %s
+WHEN MATCHED %sAND IFNULL(%s, false) = true THEN UPDATE SET %s
 WHEN NOT MATCHED THEN INSERT (%s) VALUES (%s);`,
-			// Update + Soft Deletion
-			idempotentClause, sql.BuildColumnsUpdateFragment(cols, constants.StagingAlias, constants.TargetAlias, sd),
+			// Update + soft deletion when we have previous values (update all columns)
+			idempotentClause, sql.GetQuotedOnlySetDeleteColumnMarker(constants.StagingAlias, sd), sql.BuildColumnsUpdateFragment(cols, constants.StagingAlias, constants.TargetAlias, sd),
+			// Soft deletion when we don't have previous values (only update the __artie_delete column)
+			idempotentClause, sql.GetQuotedOnlySetDeleteColumnMarker(constants.StagingAlias, sd), sql.BuildColumnsUpdateFragment([]columns.Column{columns.NewColumn(constants.DeleteColumnMarker, typing.Boolean)}, constants.StagingAlias, constants.TargetAlias, sd),
```

Comment:
Here's the new merge logic - if `OnlySetDeleteColumnMarker` is set to true, then we only update the `__artie_delete` column instead of updating all columns.